icu4c/source/i18n/number_longnames.cpp

```diff
@@ -206,21 +457,26 @@ void getMeasureData(const Locale &locale,
 
     // Grab desired case first, if available. Then grab no-case data to fill in
     // the gaps.
-    if (width == UNUM_UNIT_WIDTH_FULL_NAME && !unitDisplayCase.empty()) {
+    if (width == UNUM_UNIT_WIDTH_FULL_NAME && unitDisplayCase[0] != 0) {
         CharString caseKey;
         caseKey.append(key, status);
         caseKey.append("/case/", status);
         caseKey.append(unitDisplayCase, status);
 
         UErrorCode localStatus = U_ZERO_ERROR;
+        // TODO(icu-units#138): our fallback logic is not spec-compliant:
+        // lateral fallback should happen before locale fallback. Switch to
+        // getInflectedMeasureData after homogenizing data format? Find a unit
+        // test case that demonstrates the incorrect fallback logic (via
+        // regional variant of an inflected language?)
```

Comment:
optional:  could you move the description and have only the ticket number.

Reply:
Leaving as is for now - will consider for the future.